### PR TITLE
Add PSL to cpl export

### PR DESCRIPTION
**Pull request by PeterCaldwell**
PSL (sea level pressure) was all zero in cpl.hi files. This PR adds a method for computing PSL following the methodology described in #1432 and fixes the zero PSL problem. 

One question I have is whether we should keep calling the variable psl, or whether we should take this opportunity to switch to a more self-describing name like `p_sea_level`. Any opinions?

### CODE REVIEW

File: components/scream/src/control/surface_coupling_export.cpp
Comment (PeterCaldwell) [inline components/scream/src/control/surface_coupling_export.cpp:26]: Note to self: delete comments flagging where you changed code before integrating!

File: components/scream/src/control/surface_coupling_export.cpp
Comment (PeterCaldwell) [inline components/scream/src/control/surface_coupling_export.cpp:76]: I'm not particularly comfortable with getting surface T by extrapolating the 2 lowest midpoint points. EAMf90 computes this by assuming a lapse rate of 6.5 K/km, which is perhaps more numerically stable but is otherwise way more terrible. Assuming we want to use this formulation, should we be interpolating in Z or in pressure? Should we use column_scan or create a standalone function to do this computation rather than hardcoding it in here (where it isn't well tested)?

File: components/scream/src/share/util/scream_common_physics_functions_impl.hpp
Comment (PeterCaldwell) [inline components/scream/src/share/util/scream_common_physics_functions_impl.hpp:391]: This function should only be used by calculate_psl and some unit tests. I wasn't sure whether I should treat it differently (e.g. making it private or something) because of this.

File: components/scream/src/share/util/scream_common_physics_functions_impl.hpp
Comment (PeterCaldwell) [inline components/scream/src/share/util/scream_common_physics_functions_impl.hpp:407]: I was surprised that these functions are called from inside a loop over columns so - since we're only dealing with surface variables - these functions will only ever be called on scalars... not ScalarT=pack, but an actual scalar. It seems like this might limit these functions' usability (e.g. for writing output). More importantly, since these functions are called on scalars, I'm assuming it's fine to have as many if/else statements as I feel like. Another thing which could be a problem is perhaps I should be declaring things Real instead of ScalarT and not templating on ScalarT?

**Comment by AaronDonahue** [comment]
@PeterCaldwell , I like `p_sea_level` or even `pres_sea_level` over `psl`.  If now is the time to change it I vote for that.

File: components/scream/src/share/util/scream_common_physics_functions_impl.hpp
Comment (bartgol) [reply components/scream/src/share/util/scream_common_physics_functions_impl.hpp:407]: I think it's fine to keep templated on ScalarT. The code should not compile for a pack type, so there should be no risk of sneaking in bugs with unintentional behaviors.

That said, hard coding Real is perfectly fine too.

Comment (PeterCaldwell) [reply components/scream/src/share/util/scream_common_physics_functions_impl.hpp]: Changed to hard-coded Real in order to be more explicit about what these variables are.

**Comment by PeterCaldwell** [comment]
@AaronDonahue and @bartgol - this PR passes now for everything except restart on Weaver. Could you review it? I decided not to change PSL to p_sea_level now because I want to move on to other things and I'm still irrationally fond of the name PSL. I suspect we will change the name as part of a SCREAMv1.1 cleanup effort.

**Comment by AaronDonahue** [comment]
@tcclevenger , I added you as a reviewer just for the surface coupler interface changes.  If you want to check those.

File: components/scream/src/control/tests/sc_tests.cpp
Comment (AaronDonahue) [inline components/scream/src/control/tests/sc_tests.cpp:256]: I think we have a tag `ILEV` for when layouts are defined at the interface instead of midpoint.  So this should be `{COL, ILEV}`

File: components/scream/src/control/surface_coupling_export.cpp
Comment (AaronDonahue) [inline components/scream/src/control/surface_coupling_export.cpp:36]: Not at all important, but since I left a comment elsewhere I figured I would leave this comment too.  For code readability we could align the `=` sign for this line with all other lines above it.  Like you did with the addition of `p_int`.

**Review by AaronDonahue — COMMENTED** [review]
Looks good! Just a couple comments which should be simple fixes.

File: components/scream/src/control/tests/sc_tests.cpp
Comment (tcclevenger) [reply components/scream/src/control/tests/sc_tests.cpp:256]: That's my bad. I'll fix it.

File: components/scream/src/control/surface_coupling_export.cpp
Comment (tcclevenger) [reply components/scream/src/control/surface_coupling_export.cpp:36]: @PeterCaldwell I'll align this as well with my test fix (@AaronDonahue comment above)

**Comment by tcclevenger** [comment]
@AaronDonahue @bartgol I'll let one of y'all approve so I'm not approving my changes. SC looks good to me otherwise.

**Comment by PeterCaldwell** [comment]
I just realized that we haven't run CIME tests yet for this PR, so I just added that label...

**Review by AaronDonahue — APPROVED** [review]

**Comment by AaronDonahue** [comment]
The only FAILs in the AT for this look like the expected DIFFs in the two SCREAMv1 CIME based runs.  Given that this PR is non-BFB.  I'm going to manually merge now.

**Review by AaronDonahue — APPROVED** [review]

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (bartgol) [inline components/scream/src/share/tests/common_physics_functions_tests.cpp:71]: To be clear, the common physics functions come in two flavors: one templated on ScalarT, without assumptions on what that scalar is; and one that works on a full column of ScalarT (which simply runs a ||for loop calling the former). As such, one can call them with `Scalar=Pack<Real,N>` (as we often do), as well as with `Scalar=Real`.

That said, since these functions are always gonna be called with 2d inputs/outputs, it's perfectly fine to test them separately.

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (bartgol) [inline components/scream/src/share/tests/common_physics_functions_tests.cpp:96]: p0 (from physics/share) is not a pack. It can be, if `Scalar=Pack<Real,N>`, but it can be a normal real with `Scalar=Real`. But since we have p0 in `src/physics/share`, we should not use it, or else we create dependency on a downstream pkg.

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (bartgol) [inline components/scream/src/share/tests/common_physics_functions_tests.cpp:104]: Do you think it would add value to randomize the inputs? Perhaps over an interval that goes a tad beyond normal values (for robustness)?

File: components/scream/src/share/util/scream_common_physics_functions_impl.hpp
Comment (bartgol) [inline components/scream/src/share/util/scream_common_physics_functions_impl.hpp:402]: Beware that, sInce 0.0065 is a double (that's the default floating point type), you are performing calculations in double precision here. `T_sl` is a double, for instance, regardless of the precision of the other inputs.

Of course, when stuff is stored in `lapse` or `T_ground_tmp` it is converted back to `Real`, so the final precision is the normal scream one. But I don't know if you want the intermediate calculations do be _always_ in double precision.

Comment (bartgol) [reply components/scream/src/share/util/scream_common_physics_functions_impl.hpp:402]: In general, if you want your hard coded numbers to match the default precision of scream (i.e., Real), you should use the `sp` function. E.g.,

```
auto t_sl = T_ground + sp(0.0065) * phi_ground/gravit;
```
Or even just `Real(0.0065)`, which is what the `sp` function does internally.

Note: for integer numbers, you can also avoid using the `.`. So long as they are used in arithmetic expressions involving Real's, the integer will be promoted to a Real. If, otoh, you write

```
auto blah = 2.0 * some_real;
```
Then `some_real` is promoted to double, and the type of `blah` is `double`.

**Comment by bartgol** [comment]
I removed automerge, since I think @tcclevenger (who worked a lot on the SC class) should have a look at this before it's merged.

**Review by tcclevenger — APPROVED** [review]

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (PeterCaldwell) [reply components/scream/src/share/tests/common_physics_functions_tests.cpp:71]: The new functions added here are different than other common physics functions because they *only* operate on reals. It would be nonsensical to test them on packs because all common physics functions are called within a loop over the horizontal and these functions are also only defined for a single vertical layer. I originally had these functions templated on ScalarT but then I got worried that someone might test them on Packs, get a fail (because what does a pack even mean in this context), and spend a lot of time debugging.

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (PeterCaldwell) [reply components/scream/src/share/tests/common_physics_functions_tests.cpp:96]: Again, problem is asserting p0 is a pack instead of "can be" a pack.

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (PeterCaldwell) [reply components/scream/src/share/tests/common_physics_functions_tests.cpp:104]: I definitely don't want to randomize the input for the property tests I've added here - I'm testing specific logic branches here. I wondered about *also* adding some randomized tests, but decided that they wouldn't add much value.

File: components/scream/src/share/tests/common_physics_functions_tests.cpp
Comment (PeterCaldwell) [reply components/scream/src/share/tests/common_physics_functions_tests.cpp:96]: Not relevant to this PR, but note that p0 *is* already used in this test file so the dependency on a downstream package is already present. I'm not clear why that would be a problem.

File: components/scream/src/share/util/scream_common_physics_functions_impl.hpp
Comment (PeterCaldwell) [reply components/scream/src/share/util/scream_common_physics_functions_impl.hpp:402]: Good catch. I meant to sp that number. I'll do that.

**Comment by PeterCaldwell** [comment]
Need another re-approval b/c I did some cleanup.

**Review by tcclevenger — APPROVED** [review]

**Comment by AaronDonahue** [comment]
Just checking @PeterCaldwell that you have reviewed all of @bartgol 's comments are comfortable merging this now.  Is that true?

**Comment by PeterCaldwell** [comment]
I got a bit overexcited about making changes and am seeing build fails on Weaver and Mappy now but I can't debug them (b/c I don't have machine access and because I'm in meetings) so I don't know whether they are real or not. Check out the latest AT tests...

**Comment by bartgol** [comment]
@PeterCaldwell The fail on weaver seems to be a build fail. I can take a look.

**Comment by PeterCaldwell** [comment]
Ok, I think PSL is back to behaving as expected. Can you check over and integrate, @AaronDonahue ?

**Comment by AaronDonahue** [comment]
Looks good.  Now it's passing the weaver build and tests and the only fails I see are the expected mappy DIFFs since this is non-BFB.

**Comment by AaronDonahue** [comment]
Note, we have already merged Ben's non-BFB rrtmgp PR.  Are we comfortable merging both of these non-BFB PR's the same day?  Just doing my diligence before manually merging. @PeterCaldwell? @bartgol ?

**Comment by PeterCaldwell** [comment]
I don't think there's a rush - we can do it first thing tomorrow AM.

**Comment by tcclevenger** [comment]
@bartgol Did you have a chance to look at this? If not and you don't have time, I can real quick. It may involve SC stuff.

**Comment by AaronDonahue** [comment]
@tcclevenger and @bartgol , I was planning on manually merging this today now that it looks like it is passing the AT (except the non-BFB DIFFs that we expect).  But if you want one more look at it, I can hold off.

**Comment by tcclevenger** [comment]
Oh I thought it had a build fail on weaver. It's good to merge for me.

**Comment by PeterCaldwell** [comment]
@tcclevenger - I'm unclear what problem you suggest Luca look at - the only failures we currently see are (according to @AaronDonahue ) expected cprnc diffs on mappy due to the addition of an additional output variable. I think this is ready to merge (so long as someone takes one last glance at the code). 

I was thinking that Aaron was waiting to do this until new baselines were created for the current master, which @jgfouca is doing... but maybe I'm wrong.

**Comment by AaronDonahue** [comment]
okay cool.  I'm going to manually merge now.

**Pull request merged**
